src/ext/java/org/opentripplanner/ext/fares/impl/OrcaFareServiceImpl.java

```diff
@@ -0,0 +1,665 @@
+            fare,
+            fareType,
+            currency,
+            legFare - orcaFareDiscount,
+            orcaFareDiscount
+          );
+          orcaFareDiscount = legFare;
+        } else {
+          // Ride is free, counts as a transfer if legFare is NOT free
+          addFareComponent(leg, fare, fareType, currency, 0, legFare != 0 ? orcaFareDiscount : 0);
+        }
+      } else if (usesOrca(fareType) && !inFreeTransferWindow) {
+        // If using Orca and outside of the free transfer window, add the cumulative Orca fare (the maximum leg
+        // fare encountered within the free transfer window).
+        cost += orcaFareDiscount;
+
+        // Reset the free transfer start time and next Orca fare as needed.
+        if (ridePermitsFreeTransfers) {
+          // The leg is using a ride type that permits free transfers.
+          // The next free transfer window begins at the start time of this leg.
+          freeTransferStartTime = leg.getStartTime();
+          // Reset the Orca fare to be the fare of this leg.
+          orcaFareDiscount = legFare;
+        } else {
+          // The leg is not using a ride type that permits free transfers.
+          // Since there are no free transfers for this leg, increase the total cost by the fare for this leg.
+          cost += legFare;
+          // The current free transfer window has expired and won't start again until another leg is
+          // encountered that does have free transfers.
+          freeTransferStartTime = null;
+          // The previous Orca fare has been applied to the total cost. Also, the non-free transfer cost has
+          // also been applied to the total cost. Therefore, the next Orca cost for the next free-transfer
+          // window needs to be reset to 0 so that it is not applied after looping through all rides.
+          orcaFareDiscount = 0;
+        }
+        addFareComponent(leg, fare, fareType, currency, legFare, 0);
+      } else {
+        // If not using Orca, add the agency's default price for this leg.
+        addFareComponent(leg, fare, fareType, currency, legFare, 0);
+        cost += legFare;
+      }
+    }
+    cost += orcaFareDiscount;
+    if (cost < Float.POSITIVE_INFINITY) {
+      fare.addFare(fareType, getMoney(currency, cost));
+    }
+    return cost < Float.POSITIVE_INFINITY;
+  }
+
+  /**
+   * Adds a fare component to a given ride.
+   * @param leg Ride receiving fare component
+   * @param fareType Fare type for fare component
+   * @param totalFare Cost of leg fare after transfer
+   * @param transferDiscount Transfer discount applied or 0 if no transfer was used.
+   */
+  private static void addFareComponent(
+    Leg leg,
+    ItineraryFares itineraryFares,
+    FareType fareType,
+    Currency currency,
+    float totalFare,
+    float transferDiscount
+  ) {
+    var id = new FeedScopedId("orcaFares", "farePayment");
+    var riderCategory = new RiderCategory("orcaFares", getFareCategory(fareType), "");
```

Comment:
I re-read the draft spec for Fares V2 and the URL field is optional.

Can you do the following:

- Pass in `null` as the URL here
- Add the `@Nullable` annotation to `RiderCategory`
- Check that the API mapper deals with it correctly